File: JavaScript/696-Count Binary Substrings.js
/**
 * @param {string} s
 * @return {number}
 */
var countBinarySubstrings = function(s) {
    let prev = 0, curr = 1, res = 0;
    
    for (let i=1; i<s.length; i++){
        if (s[i] !== s[i-1]){
            res += Math.min(prev, curr);
            [prev, curr] = [curr, 1];
        }else{
            curr += 1;
        }
    }
    
    return res + Math.min(prev, curr);
};

File: C++/696-Count Binary Substrings.cpp
class Solution {
public:
    int countBinarySubstrings(string s) {
        int prev = 0, count = 0, res = 0;
        
        for (int i=0; i<s.size(); i++) {
            if (i > 0 && s[i] != s[i-1]) {
                prev = count;
                count = 1;
            } else {
                count++;
            }
            if (prev >= count) {
                res++;
            }
        }

        return res;
    }
};